src/module-elasticsuite-catalog-optimizer/Model/Optimizer/Preview.php

```diff
@@ -161,25 +161,26 @@ private function getOptimizedResults()
     }
 
     /**
-     * Indicates if the current optmizer can be applied to the search context.
+     * Indicates if the current optimizer can be applied to the search context.
      *
      * @return boolean
      */
-    private function canApply()
+    private function canApply() : bool
     {
-        $canApply = in_array($this->containerConfiguration->getName(), $this->optimizer->getSearchContainer());
+        $canApply = !empty($this->optimizer->getSearchContainer())
+            && in_array($this->containerConfiguration->getName(), $this->optimizer->getSearchContainer(), true);
 
-        if ($canApply && $this->containerConfiguration->getName() == 'quick_search_container') {
+        if ($canApply && $this->containerConfiguration->getName() === 'quick_search_container') {
             $config = $this->optimizer->getQuickSearchContainer();
-            if ($config['apply_to'] == 1) {
+            if ($config['apply_to'] === 1) {
```

Comment:
Beware with strict comparison here. Could be `'1'` and not `1` when fetched from DB. If you wanna strict-check, I'd go for `if ( (int) $config['apply_to'] === 1)` to be really sure.